community/louvain_common.go

```diff
@@ -35,11 +36,194 @@ func Q(g graph.Graph, communities [][]graph.Node, resolution float64) float64 {
 	}
 }
 
+// ReducedGraph is a modularised graph.
+type ReducedGraph interface {
+	graph.Graph
+
+	// Communities returns the community memberships
+	// of the nodes in the graph used to generate
+	// the reduced graph.
+	Communities() [][]graph.Node
+
+	// Structure returns the community structure of
+	// the current level of the module clustering.
+	// The first index of the returned value
```

Comment:
So, maybe seeing a small graph level chain will help. This is the level structure of the modularisation of the small dumbell graph. The first value is the final result of `Modularize` and the second value is the parent of that (the returned value from `Expanded()`):

```
&{
	nodes:[
		{id:0, nodes:[4 5 3], weight:6},
		{id:1, nodes:[1 0 2], weight:6},
	],
	undirectedEdges:{edges:[[1] [0]] weights:map[[0 1]:1]},

	communities:[
		[0], // Community 1.
		[1], // Community 2.
	],

	parent:0xc42042e4e0,
}

&{
	nodes:[
		{id:0, nodes:[0], weight:0},
		{id:1, nodes:[1], weight:0},
		{id:2, nodes:[2], weight:0},
		{id:3, nodes:[3], weight:0},
		{id:4, nodes:[4], weight:0},
		{id:5, nodes:[5], weight:0},
	],
	undirectedEdges:{edges:[[1 2] [0 2] [0 1 3] [4 5 2] [3 5] [3 4]] weights:map[[3 5]:1 [4 5]:1 [0 1]:1 [0 2]:1 [1 2]:1 [2 3]:1 [3 4]:1]},

	communities:[
		[4 5 3], // Members of community 1.
		[1 0 2], // Members of community 2.
	],

	parent:<nil>,
}
```

When there are more than two layers, the pattern continues. Does this help?